File: Prueba_4.py
import random

numero = random.randint(0,10)
intentos = 3

print('Adivina el número entre 0 y 10')
print(f'Tienes {intentos} intentos.')

for i in range(intentos):

    adivina = int(input("Ingresa el número: "))

    if adivina == numero:
        print('¡Adivinaste el número!')
        break
    else:
        if i < intentos -1: #Todavía quedan intentos
            if adivina < numero:
                print('El número secreto es mayor')
            else:
                print('El número secreto es  menor')
            print(f'Te quedan {intentos-(i+1)} intentos')
        
        else:
            print(f'❌ Usaste todos los intentos. El número era {numero}.')

File: Prueba4.js
const prompt = require("prompt-sync")();

let numero = Math.floor(Math.random() * 11);  // número aleatorio entre 0 y 10
let intentos = 3;

console.log('Adivina el número secreto entre 0 y 10');
console.log(`Tienes ${intentos} intentos`);

for (let i = 0; i < intentos; i++) {
    let adivina = parseInt(prompt("Escribe el número:"));

    if (adivina === numero) {
        console.log('🎉 Adivinaste el número');
        break;
    } else {
        if (i < intentos - 1) {
            if (adivina < numero) {
                console.log('El número secreto es mayor 📈');
            } else {
                console.log('El número secreto es menor 📉');
            }
            console.log(`Te quedan ${intentos - (i + 1)} intentos`);
        } else {
            console.log(`❌ Usaste todos los intentos. El número era ${numero}`);
        }
    }
}